src/components/pages/Products/useProducts.ts

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from '@tanstack/react-query'
+import { useState } from 'react'
+
+import { getProducts } from 'services/products'
+
+const useProducts = () => {
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const { isLoading, isError, data } = useQuery({
+    queryKey: ['products', currentPage],
+    queryFn: () => getProducts(currentPage),
+    keepPreviousData: true,
+  })
+
+  const products = data?.list
+  const totalPages = data?.meta.pages
+
+  const onPageChange = (pageNumber: number) => {
+    setCurrentPage(pageNumber + 1)
+    window.scrollTo({ top: 0, behavior: 'smooth' })
```

Comment:
doesn't mean for you to change it, but it may cause some problems https://caniuse.com/?search=scroll-behavior%3A%20smooth